src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +359,510 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
+                                   busType={diskParams.busType}
+                                   serial={diskParams.serial}
+                                   validationFailed={validationFailed}
+                                   supportedDiskBusTypes={supportedDiskBusTypes} />
+            </>
+        );
     }
 
-    onInsertClicked() {
-        const Dialogs = this.context;
-        const close = Dialogs.close;
-        const { vm } = this.state;
-        const { disk } = this.props;
+    return (
+        <Modal position="top" variant="medium" id={`${idPrefix}-dialog-modal-window`} isOpen onClose={Dialogs.close}
+               title={isMediaInsertion ? _("Insert disc media") : _("Add disk")}
+               footer={
+                   <AddDiskModalFooter
+                     dialogLoading={dialogLoading}
+                     dialogErrorSet={dialogErrorSet}
+                     disk={disk}
+                     diskParams={diskParams}
+                     idPrefix={idPrefix}
+                     isMediaInsertion={isMediaInsertion}
+                     mode={mode}
+                     setValidate={setValidate}
+                     storagePoolName={storagePoolName}
+                     storagePools={storagePools}
+                     validate={validate}
+                     validationFailed={_validationFailed}
+                     verificationInProgress={verificationInProgress}
+                     vm={vm}
+                     vms={vms}
+                   />
+               }
+        >
+            {dialogError && <ModalError dialogError={dialogError} dialogErrorDetail={dialogErrorDetail} />}
+            {defaultBody}
+        </Modal>
+    );
+};
 
-        const validation = this.validateParams();
-        if (Object.getOwnPropertyNames(validation).length > 0)
-            return this.setState({ addDiskInProgress: false, validate: true });
+const AddDiskModalFooter = ({
+    customDiskVerificationFailed,
+    customDiskVerificationMessage,
+    dialogErrorSet,
+    dialogLoading,
+    diskParams,
+    idPrefix,
+    isMediaInsertion,
+    mode,
+    setValidate,
+    storagePoolName,
+    storagePools,
+    validate,
+    validationFailed,
+    verificationInProgress,
+    vm,
+    vms,
+}) => {
+    const [addDiskInProgress, setAddDiskInProgress] = useState(false);
+    const Dialogs = useDialogs();
 
-        this.setState({ addDiskInProgress: true, validate: false });
+    const onInsertClicked = () => {
         return domainInsertDisk({
             connectionName: vm.connectionName,
             vmName: vm.name,
-            target: disk.target,
-            diskType: this.state.mode === CUSTOM_PATH ? "file" : "volume",
-            file: this.state.file,
-            poolName: this.state.storagePoolName,
-            volumeName: this.state.existingVolumeName,
+            target: diskParams.target,
+            diskType: mode === CUSTOM_PATH ? "file" : "volume",
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231114-134752-e3097bd0-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)